lib/ntbb-session.lib.php

```diff
@@ -475,7 +475,13 @@ function getAssertion($userid, $serverhostname, $user = null, $challengekeyid =
 		if (($user['userid'] === $userid) && !empty($user['loggedin'])) {
 			// already logged in
 			$usertype = '2';
-			if (in_array($userid, $psconfig['sysops'], true)) {
+			$isupperstaff = $curuser['group'] === 6 || $curuser['group'] === 2;
```

Comment:
probably these should all be stuck in `$curuser` instead of having a bunch more globals.

Reply:
Or maybe a separate `$curauth`.

Honestly I would design this so differently if I was writing this today...

Reply:
Oh, I misunderstood this. Yeah, this is fine.